src/decorators.py

```diff
@@ -0,0 +1,29 @@
+'''
+Copyright (C) 2020 Link Shortener Authors (see AUTHORS in Documentation).
+Licensed under the MIT (Expat) License (see LICENSE in Documentation).
+'''
+from functools import wraps
+from decouple import config
+
+from sanic.response import json
+
+
+whitelist = [
+    config('WHITELISTED_EMAIL_1'),
+    config('WHITELISTED_EMAIL_2')
+]
+
+
+def credential_whitelist_check():
+
+    def decorator(original_func):
+        @wraps(original_func)
+        async def wrapper(request, user, *args, **kwargs):
+            if (user.email.split('@')[1] in whitelist):
+                response = await original_func(request, user, *args, **kwargs)
+                return response
+            else:
+                return json({'message': 'unauthorized'}, status=401)
+        return wrapper
+
+    return decorator
```

Comment:
Just FYI, you can do the decorator with one less function like this:
```suggestion
def credential_whitelist_check(original_func):

    @wraps(original_func)
    async def wrapper(request, user, *args, **kwargs):
        if (user.email.split('@')[1] in whitelist):
            response = await original_func(request, user, *args, **kwargs)
            return response
        else:
            return json({'message': 'unauthorized'}, status=401)
    return wrapper
```
if you then use the decorator like this:
```python
@credential_whitelist_check
async def my_func():
```
instead of
```python
@credential_whitelist_check()
async def my_func():
```
(note the missing parenthesis)
but no need to redo it! Just FYI!...

Reply:
Changed